backend/danswer/db/engine.py

```diff
@@ -225,27 +230,120 @@ def get_sqlalchemy_async_engine() -> AsyncEngine:
     return _ASYNC_ENGINE
 
 
-def get_session_with_tenant(tenant_id: str | None = None) -> Session:
+# Dependency to get the current tenant ID and set the context variable
+def get_current_tenant_id(request: Request) -> str:
```

Comment:
I generally feel like it's more clear to separate this out as a None vs whatever the tenant id is. What do you think? Maybe it just introduces more repeated logic that has to be spread out everywhere.

Also is there now a tenant_public? How do we differentiate between no tenant / public schema vs tenant_public?